src/tree/registries/Azure/AzureRegistryDataProvider.ts

```diff
@@ -0,0 +1,126 @@
+                return [];
+            }
+
+            const subscriptions = await this.subscriptionProvider.getSubscriptions();
+
+            return subscriptions.map(sub => {
+                return {
+                    parent: element,
+                    label: sub.name,
+                    type: 'azuresubscription',
+                    subscription: sub,
+                    additionalContextValues: ['azuresubscription']
+                } as AzureSubscriptionRegistryItem;
+            });
+        } else if (isAzureSubscriptionRegistryItem(element)) {
+            return await this.getRegistries(element);
+        } else {
+            const children = await super.getChildren(element);
+
+            if ((element as AzureRegistryItem)?.subscription) {
+                children.forEach(e => {
+                    e.subscription = (element as AzureRegistryItem).subscription;
+                });
+            }
+
+            return children;
+        }
+    }
+
+    public dispose(): void {
+        this.subscriptionProvider.dispose();
+    }
+
+    public async getRegistries(subscriptionItem: CommonRegistryItem): Promise<AzureRegistry[]> {
+        subscriptionItem = subscriptionItem as AzureSubscriptionRegistryItem;
+        // TODO: replace this with `createAzureClient`
+        const acrClient = new (await import('@azure/arm-containerregistry')).ContainerRegistryManagementClient(subscriptionItem.subscription.credential, subscriptionItem.subscription.subscriptionId);
+
+        const registries: AcrRegistry[] = [];
+
+        for await (const registry of acrClient.registries.list()) {
+            registries.push(registry);
+        }
+
+        return registries.map(registry => {
+            return {
+                parent: subscriptionItem,
+                type: 'commonregistry',
+                baseUrl: vscode.Uri.parse(`https://${registry.loginServer}`),
+                label: registry.name!,
+                iconPath: vscode.Uri.joinPath(this.extensionContext.extensionUri, 'resources', 'azureRegistry.svg'),
+                subscription: subscriptionItem.subscription,
+            };
+        });
+    }
+
+    public override getTreeItem(element: CommonRegistryItem): Promise<vscode.TreeItem> {
+        if (isAzureSubscriptionRegistryItem(element)) {
+            return Promise.resolve({
+                label: element.label,
+                collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+                contextValue: 'azuresubscription',
+                iconPath: vscode.Uri.joinPath(this.extensionContext.extensionUri, 'resources', 'azureSubscription.svg'),
+            });
+        } else {
+            return super.getTreeItem(element);
+        }
+    }
+
+    protected override getAuthenticationProvider(item: AzureRegistryItem): ACROAuthProvider {
+        const registryString = item.registryUri.toString();

```

Comment:
Isn't `registryUri` removed in favor of `baseUrl`?

Reply:
yeah you're right. this is fixed in branch `yangalex/azureRegistries`, you should see the changes in that PR

Reply:
let em add the changes here too